### ISSUE TITLE
is_allowed improvement

### ISSUE THREAD

**Issue by stupidoes** [emreakay/CodeIgniter-Aauth#166]
i found dead code here : 

```
if( $user_id===FALSE){
    return $this->is_group_allowed($perm_id);
} 
```

since $user_id is already set to get from session above. Even when user not logged in, $user_id would become NULL instead.

```
if( $user_id == FALSE){
    $user_id = $this->CI->session->userdata('id');
}
```

and i think, it's better to put 'break' when found permission inside group since it wouldn't need to check another group for the same permission.

```
foreach( $this->get_user_groups($user_id) as $group ){
    if ( $this->is_group_allowed($perm_id, $group->id) ){
        $g_allowed=TRUE;
        break;
    }
}
```

**Comment by REJack** [comment]
Hi @stupidoes,

in my opinion the first 2 codeblocks are not dead code, only this can check perms for the public group, if we delete this then the public group is useless.

the idea with the break is good, i change it later and create a new release for this.

**Comment by stupidoes** [comment]
Could you provide me with case when $user_id is still in FALSE value when entering the condition ? Because i think it will become $this->CI->session->userdata('id') value or NULL if user not logged in.

**Comment by REJack** [comment]
:dizzy_face:  sorry you are right the "dead code" is really dead and its not needed too.

**Comment by stupidoes** [comment]
Thanks.. :)

**Comment by REJack** [comment]
I close this issue 😄 
Enhancement is released with [v2.5.12](https://github.com/emreakay/CodeIgniter-Aauth/releases/tag/v2.5.12)
